packages/components/src/Actions/ActionFile/ActionFile.component.js

```diff
@@ -12,46 +12,6 @@ import theme from './ActionFile.scss';
 const LEFT = 'left';
 const RIGHT = 'right';
 
-function getIcon({ icon, iconTransform, inProgress }) {
-	if (inProgress) {
-		return <CircularProgress size="small" key="icon" />;
-	}
-
-	if (icon) {
-		return <Icon name={icon} transform={iconTransform} key="icon" />;
-	}
-
-	return null;
-}
-getIcon.propTypes = {
-	icon: PropTypes.string,
-	iconTransform: PropTypes.string,
-	inProgress: PropTypes.bool,
-};
-
-function getLabel({ hideLabel, label }) {
-	if (hideLabel) {
-		return null;
-	}
-	return <span key="label">{label}</span>;
-}
-
-getLabel.propTypes = {
-	label: PropTypes.string,
-	hideLabel: PropTypes.bool,
-};
-
-function adjustContentPlacement(icon, label, iconPosition) {
-	if (iconPosition === RIGHT) {
-		return [label, icon];
-	}
-	return [icon, label];
-}
-
-function getButtonContent(props) {
-	return adjustContentPlacement(getIcon(props), getLabel(props), props.iconPosition);
-}
-
```

Comment:
The lint problem was that some props are defined but not used. In fact they are used in those function, passing.... **all** the props...

The other problem was readability. This split of logic is a little too much, some functions can totally be inlined. That makes developers scroll a lot, changing the context everytime we want to look at a new function.

* I replaced that with the 6 lines of code in render. That fixed the lint errors as the props are explicitly used in the component
* fixed the proptypes